platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -371,6 +372,7 @@ public Set<Configuration> getExtendsFrom() {
     @Override
     public Configuration setExtendsFrom(Iterable<Configuration> extendsFrom) {
         validateMutation(MutationType.HIERARCHY);
+        assertNotDetachedExtensionDoingExtending(extendsFrom);
```

Comment:
We also call `extendsFrom(Configuration...)` from this method.

We should either remove this line and allow the `extendsFrom(configuration)` to report the deprecated action, or wrap the `extendsFrom(configuration)` line below in a `DeprecationLogger.whileDisabled`

Reply:
It shouldn't matter, as (currently) the deprecation logger will automatically de-duplicate and only log the message once and (in Gradle 9.0) the exception will be thrown by the call into the assertion method, precluding the second call.  

Just as we call `validateMutation` twice, we should call this assertion method twice, to be consistent with the pattern.